app/services/music_import_service/recording_collector.rb

```diff
@@ -0,0 +1,334 @@
+    @found_bibs += 1
+  rescue StandardError
+    logger.info "Errored trying to populate bib."
+  end
+
+  # SQL QUERIES
+  def recordings_query
+    "select R.idRecording, R.CallNo, R.RecTitle, C.CourseNo from Recordings R " \
+      "left join Selections S on S.idRecording=R.idRecording " \
+      "left join jSelections jS on S.idSelection=jS.idSelection " \
+      "left join Courses C on jS.idCourse=C.idCourse"
+  end
+
+  def bib_query(column:, call_num:)
+    "SELECT bibid, title from orangelight_call_numbers where #{column}='#{call_num}'"
+  end
+
+  def bib_numbers_for(call_number:)
+    return [] unless call_number
+    bib_numbers = []
+    normalization_strategies.each_pair do |strategy, column|
+      next unless bib_numbers.empty?
+      normalized = send(strategy, call_number)
+      bib_numbers = query_ol(column: column, call_number: normalized)
+      logger.info "Found bib with #{strategy} for #{call_number}" unless bib_numbers.empty?
+    end
+    bib_numbers
+  end
+
+  def normalization_strategies
+    { space_after_hyphen: "label",
+      space_replace_hyphen: "label",
+      space_after_hyphen_lc: "sort",
+      space_replace_hyphen_lc: "sort",
+      just_lc: "sort",
+      volume_expansion: "label",
+      volume_space_expansion: "label" }
+  end
+
+  # check the call numbers database for the call number given
+  # If multiple call numbers are found, gets all the bibs from the ol API
+  # @return Array of bib id strings, may be empty
+  def query_ol(column:, call_number:)
+    result = postgres_adapter.execute(query: bib_query(column: column, call_num: call_number))
+    return [] if result.empty?
+    if result.first["title"].match?(/^[\d]+ titles with this call number$/)
+      logger.info "  trying ol"
+      bib_numbers = query_ol_api(call_number: call_number)
+    else
+      bib_numbers = result.map { |h| h["bibid"] }.uniq
+    end
+    bib_numbers
+  end
+
+  # query orangelight as a backup for when the database returns a querystring param
+  #   "label"=>"CD- 9221",
+  #   e.g. "?f[call_number_browse_s][]=CD-+9221"
+  def query_ol_api(call_number:)
+    conn = Faraday.new(url: "https://catalog.princeton.edu/catalog.json")
+    result = conn.get do |req|
+      req.params["search_field"] = "all_fields"
+      req.params["f[call_number_browse_s][]"] = call_number
+    end
+
+    json = JSON.parse result.body
+    # return nil unless json["response"]["docs"].count.positive?
+    json["response"]["docs"].map { |doc| doc["id"] } # will be [] if no results
+  end
+
+  # apply normalization which replaces a space with a hyphen
+  def space_after_hyphen(call_number)
+    return unless call_number
+    general_normalization(call_number) do |cn|
+      # dashes should be followed by spaces e.g. cd-10994
+      # ls, cass, dat, cd, vcass, dvd
+      if cn.match?(/^((CD)||(DAT)||(CASS)||(LS)||(VCASS)||(DVD))-/)
```

Comment:
This is repeated a bunch of times — extract to method?

Reply:
I'll do it this